opentelemetry-appender-tracing/src/layer.rs

```diff
@@ -1,60 +1,122 @@
-use opentelemetry::logs::{LogRecord, Logger, LoggerProvider, Severity};
+use opentelemetry::{
+    logs::{AnyValue, LogRecord, Logger, LoggerProvider, Severity},
+    Key,
+};
 use std::borrow::Cow;
-use tracing_core::Level;
+use tracing_core::{Level, Metadata};
+use tracing_log::NormalizeEvent;
 use tracing_subscriber::Layer;
 
 const INSTRUMENTATION_LIBRARY_NAME: &str = "opentelemetry-appender-tracing";
+// Start by "log."
+const LOG_METADATA_ATTRIBUTES: &[&str] = &[
+    "name",
+    "target",
+    "module.path",
+    "file.path",
+    "file.name",
+    "file.line",
+    "module_path",
+    "file",
+    "line",
+];
 
 /// Visitor to record the fields from the event record.
-struct EventVisitor<'a> {
-    log_record: &'a mut LogRecord,
+#[derive(Default)]
+struct EventVisitor {
+    log_record_attributes: Vec<(Key, AnyValue)>,
+    log_record_body: Option<AnyValue>,
 }
 
-impl<'a> tracing::field::Visit for EventVisitor<'a> {
+fn is_metadata(field: &str) -> bool {
+    if let Some(log_field) = field.strip_prefix("log.") {
```

Comment:
Why to strip `log.` prefix - the metadata from log/tracing crate won't be having it anyway. Or am I missing something :)

Reply:
A little quirk from me :p. All the metadata we have for the log lines begin with 'log.' so I though that checking if the name begins with 'log.' is quicker than always looping the LOG_METADATA_ATTRIBUTES array. But honestly I can remove it if it is confusing. :)